src/tools/PICS-generator/XMLPICSValidator.py

```diff
@@ -0,0 +1,122 @@
+from spec_parsing_support import build_xml_clusters  # noqa: E402
+
+console = None
+xml_clusters = None
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--pics-template', required=True)
+args, unknown = parser.parse_known_args()
+
+xmlTemplatePathStr = args.pics_template
+if not xmlTemplatePathStr.endswith('/'):
+    xmlTemplatePathStr += '/'
+
+# Load PICS XML templates
+print("Capture list of PICS XML templates")
+xmlFileList = os.listdir(xmlTemplatePathStr)
+for PICSXmlFile in xmlFileList:
+    print(f"{xmlTemplatePathStr}/{PICSXmlFile}")
+
+
+class XMLPICSValidator(MatterBaseTest):
+    @async_test_body
+    async def test_xml_pics_validator(self):
+
+        # Create console to print
+        global console
+        console = Console()
+
+        global xml_clusters
+        xml_clusters, problems = build_xml_clusters()
+
+        for cluster in xml_clusters:
+
+            fileName = ""
+            clusterName = xml_clusters[cluster].name
+
+            if "ICDManagement" == clusterName:
+                picsFileName = "ICD Management"
+
+            elif "OTA Software Update Provider" in clusterName or \
+                 "OTA Software Update Requestor" in clusterName:
+                picsFileName = "OTA Software Update"
+
+            elif "On/Off" == clusterName:
+                picsFileName = clusterName.replace("/", "-")
+
+            elif "GroupKeyManagement" == clusterName:
+                picsFileName = "Group Communication"
+
+            elif "Wake on LAN" == clusterName or \
+                 "Low Power" == clusterName or \
+                 "Keypad Input" == clusterName or \
+                 "Audio Output" == clusterName or \
+                 "Media Input" == clusterName or \
+                 "Target Navigator" == clusterName or \
+                 "Content Control" == clusterName or \
+                 "Channel" == clusterName or \
+                 "Media Playback" == clusterName or \
+                 "Account Login" == clusterName or \
+                 "Application Basic" == clusterName or \
+                 "Content Launcher" == clusterName or \
+                 "Content App Observer" == clusterName or \
+                 "Application Launcher" == clusterName:
+
+                picsFileName = "Media Cluster"
+
+            elif "Operational Credentials" == clusterName:
+                picsFileName = "Node Operational Credentials"
+
+            # Workaround for naming colisions with current logic
+            elif "Thermostat" == clusterName:
+                picsFileName = "Thermostat Cluster"
+
+            elif "Boolean State" == clusterName:
+                picsFileName = "Boolean State Cluster"
+
+            elif "AccessControl" in clusterName:
+                picsFileName = "Access Control Cluster"
+
+            else:
+                picsFileName = clusterName
+
+            for file in xmlFileList:
```

Comment:
This could be extracted in a method, so that the build of the xmlFile list and the usage are not this separated. I would imagine we should have a global (since this seems to be the sharing method here ... not ideal) `XmlFileListLoader` that has an initial Load to construct the list and then a `findFileNameForPrefix` that does this loop.